sdk/scheduler/src/main/java/com/mesosphere/sdk/debug/PlansTracker.java

```diff
@@ -0,0 +1,297 @@
+
+    for (PlanManager planManager : planCoordinator.getPlanManagers()) {
+      Plan plan = planManager.getPlan();
+
+      //Filter down to a plan if specified.
+      if (allowFiltering && filterPlan != null && !plan.getName().equalsIgnoreCase(filterPlan)) {
+        continue;
+      }
+
+      JSONObject planObject = new JSONObject();
+      JSONArray phaseArray = new JSONArray();
+      // SUPPRESS CHECKSTYLE MultipleStringLiteralsCheck
+      planObject.put("name", plan.getName());
+      if (!skipDetails) {
+        // SUPPRESS CHECKSTYLE MultipleStringLiteralsCheck
+        planObject.put("status", plan.getStatus());
+        // SUPPRESS CHECKSTYLE MultipleStringLiteralsCheck
+        planObject.put("strategy", plan.getStrategy().getName());
+      }
+      planObject.put("phases", phaseArray);
+
+      if (!skipDetails) {
+        //Get a rollup aggregation of the steps.
+        int totalSteps = plan.getChildren().stream()
+            .flatMap(phase -> phase.getChildren().stream())
+            .collect(Collectors.toSet())
+            .size();
+        int completedSteps = plan.getChildren().stream()
+            .flatMap(phase -> phase.getChildren().stream())
+            .filter(step -> step.isComplete())
+            .collect(Collectors.toSet())
+            .size();
+
+        planObject.put("total-steps", totalSteps);
+        planObject.put("completed-steps", completedSteps);
+      }
+
+      //Iterate over phases.
+      for (Phase phase : plan.getChildren()) {
+        //Filter down to a phase if specified.
+        if (allowFiltering &&
+            filterPhase != null &&
+            !phase.getName().equalsIgnoreCase(filterPhase))
+        {
+          continue;
+        }
+
+        JSONObject phaseObject = new JSONObject();
+        JSONArray stepArray = new JSONArray();
+
+        phaseObject.put("name", phase.getName());
+        if (!skipDetails) {
+          phaseObject.put("status", phase.getStatus());
+          phaseObject.put("strategy", phase.getStrategy().getName());
+        }
+        phaseObject.put("steps", stepArray);
+
+        //Iterate over steps.
+        // SUPPRESS CHECKSTYLE NestedForDepthCheck
+        for (Step step : phase.getChildren()) {
+          //Filter down to a step if specified.
+          if (allowFiltering &&
+              filterStep != null &&
+              !step.getName().equalsIgnoreCase(filterStep))
+          {
+            continue;
+          }
+
+          JSONObject stepObject = new JSONObject();
+          stepObject.put("name", step.getName());
+          if (!skipDetails) {
+            stepObject.put("status", step.getStatus());
+            stepObject.put("errors", step.getErrors());
+          }
+
+          stepArray.put(stepObject);
+        }
+
+        phaseArray.put(phaseObject);
+      }
+
+      plansArray.put(planObject);
+    }
+    outcome.put("plans", plansArray);
+    return outcome;
+  }
+
+  // SUPPRESS CHECKSTYLE ReturnCountCheck
```

Comment:
I would love to not have this suppression!    I'm not a fan of this many returns... or put another way... I love this checkstyle :)